Makefile

```diff
@@ -113,7 +113,7 @@ PGXS := $(shell $(PG_CONFIG) --pgxs)
 EXTRA_CLEAN = $(EXT_SQL_FILE) $(DEPS)
 
 include $(PGXS)
-override CFLAGS += -DINCLUDE_PACKAGE_SUPPORT=0 -MMD -DEXT_GIT_COMMIT=\"$(EXT_GIT_COMMIT)\"
+override CFLAGS += -DINCLUDE_PACKAGE_SUPPORT=0 -MMD -DEXT_GIT_COMMIT=\"$(EXT_GIT_COMMIT)\" -DTS_FLEXIBLE_ARRAY_MEMBER=0
```

Comment:
Why is this necessary? shouldn't #define TS_FLEXIBLE_ARRAY_MEMBER FLEXIBLE_ARRAY_MEMBER work in *unix?

Reply:
clang gave me a warning without it:
```
src/hypertable_cache.c:17:13: warning: field 'q' with variable sized type 'CacheQuery'
      (aka 'struct CacheQuery') not at the end of a struct or class is a GNU extension
      [-Wgnu-variable-sized-type-not-at-end]
        CacheQuery      q;
```

Reply:
It seems like it's defined on my system to be blank rather than 0.

Reply:
You've actually discovered a bug. The issue is that CacheQuery is not the last member of HypertableCacheQuery, which it needs to be if it has a flexible array member. And HypertableCacheQuery needs to be allocated with space to account for that embeded struct.

So, we should fix the bug first and then you need not define the separate TS_FLEXIBLE_ARRAY_MEMBER and instead we can just use FLEXIBLE_ARRAY_MEMBER

Reply:
That's what I thought but then I moved it to the end and broke a bunch of the tests and someone suggested FLEXIBLE_ARRAY_MEMBER. But I will try and fix those errors and move it to the end.

Reply:
We can probably replace that data array with just an `void *data` pointer and let whoever wants to pass on data allocate that on the heap. In fact, since we now only have the hypertable cache I don't think the data field is even used. But might be useful to keep as a `void *` pointer for future.

Reply:
So, CacheQuery actually needs to be first in HypertableCacheQuery since it is casting the "derived" object to the base type (CacheQuery). The solution is to do the above thing with simply getting rid of the flexible data array and just turn it into a void pointer.

Reply:
Yeah that solution is a lot simpler and compiles on both. Thanks!